pom.xml

```diff
@@ -138,6 +139,13 @@
                 <artifactId>json</artifactId>
                 <version>20160810</version>
             </dependency>
+            <dependency>
+                <groupId>org.projectlombok</groupId>
```

Comment:
Hi @a1shadows , Great work! 

It's preferred to go with simple getter/setter methods, without using `lombok`.
This has caused IDE dependency, readability and other issues in the past.
Can you please remove this ?

Rest everything looks amazing. I might add couple of more scenarios to the test pack before merging.

Reply:
@authorjapps 

I added Lombok to reduce the boilerplate code. Are you sure we want to write explicit getters, setters and constructors for everything? 

At my workplace, we have been using Lombok extensively and at great scale with no issues faced so far, and the code stays compact.

Let me know either way and I'll make the changes

Reply:
@authorjapps  could you let me know on the above point?  Also, can you let me know when the PR looks ready to merge? Would love to see a merge commit on my name ? 😁

Reply:
@authorjapps  Did you have time to consider Lombok use in the project? We can whittle down the volume of code significantly in the long run if we adopt it. Let me know.

Reply:
@authorjapps  Hey, I have removed the Lombok dependency and fixed the test. Can you take another look at this?

Reply:
> @authorjapps Hey, I have removed the Lombok dependency and fixed the test. Can you take another look at this?

Thanks, just given a rerun now on the CI job. Once Green, we should be able to merge.

Reply:
@authorjapps  Looks like CI is a success. Can we merge now?